File: examples/features/workflow-scenario/python/cott_bindings/curriculum/workflow_scenario/request_save.py
from __future__ import annotations

from cott_runtime import U64
from curriculum.workflow_scenario_types import SaveSnapshot, SaveStatus_Queued


def request_save(snapshot: SaveSnapshot, revision: U64, text: str) -> SaveSnapshot:
    if revision <= snapshot.revision:
        return snapshot
    return SaveSnapshot(revision=revision, text=text, status=SaveStatus_Queued())

File: examples/kotlin/features/workflow-scenario/kotlin/cott_impl/curriculum/workflow_scenario/request_save.kt
package cott_impl.curriculum.workflow_scenario

internal fun request_save(snapshot: curriculum.workflow_scenario.SaveSnapshot, revision: kotlin.ULong, text: kotlin.String): curriculum.workflow_scenario.SaveSnapshot {
    require(revision > 0uL)
    if (revision <= snapshot.revision) return snapshot
    return curriculum.workflow_scenario.SaveSnapshot(
        revision = revision,
        text = text,
        status = curriculum.workflow_scenario.SaveStatus.Queued
    )
}
